pkg/storkctl/clusterpair.go

```diff
@@ -476,51 +680,76 @@ func getConfig(configFile string) clientcmd.ClientConfig {
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(configLoadingRules, configOverrides)
 }
 
-func getClusterPairParams(config, endpoint string, customPort string) (string, string, string, error) {
-	var ip, port, token string
+func getPXEndPointDetails(config string) (string, error) {
+	var ep string
 	client, err := core.NewInstanceFromConfigFile(config)
 	if err != nil {
-		return ip, port, token, err
+		return ep, err
 	}
 
 	services, err := client.ListServices("", meta.ListOptions{LabelSelector: "name=portworx-api"})
 	if err != nil || len(services.Items) == 0 {
-		err := fmt.Errorf("unable to retrieve portworx-api service from DR cluster. Err: %v", err)
-		return ip, port, token, err
+		err := fmt.Errorf("unable to retrieve portworx-api service. Err: %v", err)
+		return ep, err
 	}
-	// TODO: in case of setting up aync-dr over cloud,
-	// users set up different service as load-balancer over px apis
-	// accept px-service name as env variable
+
+	// Currently we are handling only the case where portworx-api service is a LoadBalancer
+	// For other kinds of service type like ClusterIP or NodePort, User has to provide the endpoints.
 	svc := services.Items[0]
-	ip = endpoint
-	if ip == "" {
-		// this works only if px service is converted as load balancer type
-		// TODO: for 2 cluster where worker nodes are reachable, figure out
-		// any one worker ip by looking at px/enabled label
-		ip = svc.Spec.LoadBalancerIP
-	}
-	pxToken := os.Getenv("PX_AUTH_TOKEN")
-	if len(customPort) > 0 {
-		port = customPort
-	} else {
+	// Only if px svc is load balancer type get the IP or Host details
+	if svc.Spec.Type == v1.ServiceTypeLoadBalancer {
+		if len(svc.Status.LoadBalancer.Ingress) > 0 {
+			if len(svc.Status.LoadBalancer.Ingress[0].Hostname) > 0 {
+				ep = svc.Status.LoadBalancer.Ingress[0].Hostname
+			} else if len(svc.Status.LoadBalancer.Ingress[0].IP) > 0 {
+				ep = svc.Status.LoadBalancer.Ingress[0].IP
+			}
+		}
+
 		for _, svcPort := range svc.Spec.Ports {
 			if svcPort.Name == "px-api" {
-				port = strconv.Itoa(int(svcPort.Port))
+				port := strconv.Itoa(int(svcPort.Port))
+				ep = net.JoinHostPort(ep, port)
 				break
 			}
 		}
+	} else {
+		return ep, fmt.Errorf("needed px endpoint details for service type [%s]", svc.Spec.Type)
 	}
-	pxEndpoint := net.JoinHostPort(ip, port)
+	return ep, nil
+}
+
+func getPXToken(config, pxEndpoint string) (string, error) {
+	var token string
+
+	net.SplitHostPort(pxEndpoint)
+
+	pxToken := os.Getenv("PX_AUTH_TOKEN")
```

Comment:
We can read the k8s secret called px-admin-token in portworx or kube-system namespace and fetch this token from it. Determining the namespace is going to be tough since storkctl wont know which namespace to look into. We can add --px-namespace as as a new flag with default set to portworx

Reply:
Or we list all the secrets and find the one with px-admin-token name

Reply:
File a separate ticket to handle this if you are not planning to change it as part of this PR.

Reply:
There is a separate ticket. Will put the changes as part of that PR.